### PR TITLE
Numba Ahead of Time Compilation

### PR DESCRIPTION
**Pull request by Ban-zee**
Working on precompiling numba methods. Using Numba AOT instead of jit would make me undo several changes that I have made across the duration of GSoC which might lead to nasty breaks in the code. Therefore, my plan is to include both AOT and JIT in the code for now and implement the changes gradually as the mentors see fit.

### CODE REVIEW

File: arviz/aot.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from numba.pycc import CC
+import numpy as np
+import numba
+
+module = CC("numba-compilations")
```
Comment (ahartikainen) [inline arviz/aot.py]: `module = CC("numba-compilations")` --> `module = CC("numba_compilations")`

**Comment by Ban-zee** [comment]
I guess that this pr can be merger as the only thing left is the upload of the binary extension to pypa

**Comment by canyon289** [comment]
Can you add documentation on how to use this code, upload to pypi etc? Some of the instructions may need to be added to our release wiki

**Comment by Ban-zee** [comment]
Nothing much, the user just has to use the following command :
pip install -i https://test.pypi.org/simple/ aot-arviz[it is on test pypi as of now, I'll shift it to pypa later]

This script is of no use as the package is already up on test pypi, I have included it just as backup

**Comment by canyon289** [comment]
Oh I get the confusion now. I'm not asking how a user will install it. I'm asking how do we as the maintainers compile and upload a new version ti pypi? Where does the compilation happen upon a new release?

**Comment by ahartikainen** [comment]
I think we can keep still open for now (let's make one release before this).

**Comment by canyon289** [comment]
@Ban-zee Is this PR still relevant? Or was it included in another PR that we merged during your GSOC?

**Comment by Ban-zee** [comment]
Not really very relevant but something that I'll pick up in a few months

**Comment by canyon289** [comment]
Checking in on old PRs. Any possibility of bringing this over the line or should we close?

**Comment by canyon289** [comment]
Closing for now. If this needs to be reopened feel free to do so!

**Pull request closed without merging**